File: Q31 Longest common substring.js
var longestCommonPrefix = function(arr) {
    if (!arr || arr.length === 0) {
        return "";
    }

    arr.sort();
    let a = arr[0];
    let b = arr[arr.length - 1];
    let i = 0;
    let ans = "";

    while (i < a.length) {
        if (a[i] === b[i]) {
            ans += a[i];
            i++;
        } else {
            break;
        }
    }

    return ans;
};

File: Q31 longest common substring.java


class Solution {
    public String longestCommonPrefix(String[] arr) {
        if (arr == null || arr.length == 0) {
            return "";
        }

        Arrays.sort(arr);
        String a = arr[0];
        String b = arr[arr.length - 1];
        int i = 0;
        StringBuilder ans = new StringBuilder();

        while (i < a.length()) {
            if (a.charAt(i) == b.charAt(i)) {
                ans.append(a.charAt(i));
                i++;
            } else {
                break;
            }
        }

        return ans.toString();
    }

  
}
